docs/assembly.rst

```diff
@@ -313,7 +313,9 @@ Furthermore, if the assembly block assigns to Solidity variables in memory, you
 the Solidity variables only access these memory ranges.
 
 Since this is mainly about the optimizer, these restrictions still need to be followed, even if the assembly block
-reverts or terminates. As an example, the following assembly snippet is not memory safe:
+reverts or terminates. As an example, the following assembly snippet is not memory safe in general, because the value of
+returndatasize() may exceed the 64 byte scratch space (however, if the returndatasize() is guaranteed to be less than 64,
+then this snippet would be memory safe):
```

Comment:
Yeah, that's the correct reason for this to be non-memory-safe...
However, I'm a bit wary about the proviso in the parentheses... while this is true and perfectly fine, short of actually doing ``if iszero(lt(returndatasize(), 64)) { revert(0, 0) }``, there is rarely much of a "guarantee" that ``returndatasize()`` is less than 64...

So maybe, keeping this simpler is actually better? E.g. something like this:

```suggestion
reverts or terminates. As an example, the following assembly snippet is not memory safe, because the value of
``returndatasize()`` may exceed the 64 byte scratch space:
```